src/test/java/HMW17.java

```diff
@@ -0,0 +1,55 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+
+public class HMW17 extends BaseTest {
+
+    @Test
+    public void addSongToPlaylist(){
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        WebDriver driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+        WebElement emailInput = driver.findElement(By.cssSelector("[type='email']"));
+        WebElement passwordInput = driver.findElement(By.cssSelector("[type='password']"));
+        WebElement loginButton = driver.findElement(By.cssSelector("[type='submit']"));
+        emailInput.sendKeys("vasile.tulbu@testpro.io");
+        passwordInput.sendKeys("ZgPSBpAE");
+        loginButton.click();
+        WebElement allSongsBtn=driver.findElement(By.xpath("//li/a[@href='#!/songs']"));
+        allSongsBtn.click();
+        WebElement firstSong=driver.findElement(By.xpath("//td[contains(text(), 'M33 Project - Emotional Soundtrack')]"));
+        firstSong.click();
+        WebElement addToBtn= driver.findElement(By.cssSelector("[class='btn-add-to']"));
+        addToBtn.click();
+      //  WebElement createNewPlayList= driver.findElement(By.xpath("//*[@id=\"queueWrapper\"]/header/div[3]/div/section[2]/form/input"));
+      //  createNewPlayList.sendKeys("Vasya");
+        WebElement saveBtn= driver.findElement(By.xpath("//*[@id=\"songsWrapper\"]/header/div[3]/div/section[2]/form/button"));
+        saveBtn.click();
+       // Assert.assertEquals();
+
+       // WebElement favoriteLink= driver.findElement(By.xpath("//ul/li[@class='favorites']"));
+        //favoriteLink.click();
+
+        driver.quit();
+
+
+      //  WebElement createNewPlayList= driver.findElement(By.cssSelector("[title='Create a new playlist']"));
+      //  createNewPlayList.click();
+      //  WebElement newPlayList= driver.findElement(By.xpath("//*[@id=\"playlists\"]/nav/ul/li[1]"));
+
+
+
+    }
```

Comment:
**Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```

Please, add Assert

**One test - One Assert**


**HINT**

**Step 1**

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="TestNG">
    <!-- Setting a parameter named "baseUrl" with the value "https://qa.koel.app". -->
    <parameter name="baseUrl" value="https://qa.koel.app"/>
    <test name="Example Test" preserve-order="false">
        <!-- Including a class named "Homework17" in the test. -->
        <classes>
            <class name="Homework17"/>
        </classes>
    </test>
</suite>
```

![image](https://github.com/Levise188/A52-QA-Automation-Framework/assets/104761482/8b03ff49-5b8e-4cbc-8339-788fda7df1e4)


**Step 2**

![image](https://github.com/Levise188/A52-QA-Automation-Framework/assets/104761482/4a8e2bea-fb39-4934-9dcc-4836674b3983)


```
public class BaseTest {
    protected WebDriver driver = null;


    // Setting up WebDriverManager for Chrome before the test suite runs.
    @BeforeSuite
    static void setupClass() {
        WebDriverManager.chromedriver().setup();
    }

    // Method to set up WebDriver with Chrome options before each test method.
    @BeforeMethod
    @Parameters("baseUrl")
    public void setUpDriver(String baseUrl) {
        // Configuring Chrome options for local testing.
        ChromeOptions optionsChromeLocal = new ChromeOptions();
        optionsChromeLocal.addArguments("--disable-notifications", "--remote-allow-origins=*", "--incognito", "--start-maximized", "-lang=en");
        driver = new ChromeDriver(optionsChromeLocal);
        // Configuring implicit wait for the driver and navigating to the specified URL.
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get(baseUrl);
    }

    // Method to close the browser after each test method.
    @AfterMethod
    public void closeBrowser() {
        driver.quit();
    }

    // Method for performing login action with provided email and password.
    public void login(String email, String password) {
        // Locating email, password, and login button elements and performing login action.
        WebElement emailInput = driver.findElement(By.cssSelector("[type='email']"));
        WebElement passwordInput = driver.findElement(By.cssSelector("[type='password']"));
        WebElement loginButton = driver.findElement(By.cssSelector("[type='submit']"));
        emailInput.sendKeys(email);
        passwordInput.sendKeys(password);
        loginButton.click();
    }
}
```


**Step 3**

![image](https://github.com/Levise188/A52-QA-Automation-Framework/assets/104761482/4efd63a5-182c-47e6-ba29-af214ff895f3)


```
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.testng.Assert;
import org.testng.annotations.Test;

// Homework17 class extends BaseTest for Selenium test automation.
public class Homework17 extends BaseTest {

    // Test method for adding a song to a playlist.
    @Test
    public void addSongToPlaylist() throws InterruptedException {
        // GIVEN
        // Initialization of test data.
        String nameSong = "pluto";
        // Write the current name of your playlist
        String namePlaylist = "TestPro Playlist";
        String expectedSongAddedMessage = "Added 1 song into \"" + namePlaylist + ".\"";

        // WHEN
        // Performing actions: login, search for a song, select the first result, add to playlist.
        login("demo@class.com", "te$t$tudent");
        Thread.sleep(2000);
        searchSong(nameSong);
        clickViewAllBtn();
        selectFirstSongResult();
        clickAddToBtn();
        choosePlaylist(namePlaylist);

        // THEN
        // Asserting that the expected success message is displayed.
        Assert.assertEquals(getAddToPlaylistSuccessMsg(), expectedSongAddedMessage);
    }

    // Method to search for a song by name.
    public void searchSong(String nameSong) throws InterruptedException {
        // Locating the search field and entering the song name.
        WebElement searchField = driver.findElement(By.cssSelector("div#searchForm input[type='search']"));
        searchField.sendKeys(nameSong);
        Thread.sleep(2000);
    }

    // Method to click the "View All" button.
    public void clickViewAllBtn() throws InterruptedException {
        // Locating and clicking the "View All" button.
        WebElement viewAll = driver.findElement(By.xpath("//button[@data-test='view-all-songs-btn']"));
        viewAll.click();
        Thread.sleep(2000);
    }

    // Method to select the first song result.
    public void selectFirstSongResult() throws InterruptedException {
        // Locating and clicking the first song result.
        WebElement firstSong = driver.findElement(By.xpath("//section[@id='songResultsWrapper']//tr[@class='song-item'][1]"));
        firstSong.click();
        Thread.sleep(2000);
    }

    // Method to click the "Add To" button.
    public void clickAddToBtn() throws InterruptedException {
        // Locating and clicking the "Add To" button.
        WebElement addToButton = driver.findElement(By.xpath("//section[@id='songResultsWrapper']//button[@data-test='add-to-btn']"));
        addToButton.click();
        Thread.sleep(2000);
    }

    // Method to choose a playlist by name.
    public void choosePlaylist(String playlistName) throws InterruptedException {
        // Locating and clicking the playlist with the specified name.
        WebElement playlist = driver.findElement(By.xpath("//*[@id='songResultsWrapper']//*[contains(text(), '" + playlistName + "')]"));
        playlist.click();
        Thread.sleep(2000);
    }

    // Method to get the success message after adding to the playlist.
    public String getAddToPlaylistSuccessMsg() {
        // Locating and retrieving the success message.
        WebElement notification = driver.findElement(By.cssSelector("div.success.show"));
        return notification.getText();
    }
}
```

![1](https://github.com/Levise188/A52-QA-Automation-Framework/assets/104761482/98de737c-9285-4a4b-a308-58f044e5c7bc)

**RESULT:**

![image](https://github.com/Levise188/A52-QA-Automation-Framework/assets/104761482/c1dc337e-4a46-41ab-b937-59a1f94c4633)

Reply:
Thank you, Alex; your suggestion makes more sense; the only thing is that you used a lot of things that I didn’t know existed. I was sure that should be something like this, but I didn’t know what syntax to use. I will use your suggestion for the code. Thank you